src/components/PriorityBadge/PriorityBadge.tsx

```diff
@@ -0,0 +1,245 @@
+                        onStateChange: (state) => toggleDropdown('priority', state.isOpen),
+                    }}
+                    onChange={(newPriority) => onPriorityChange(newPriority)}
+                    placement="bottom-start"
+                    refElement={badgeRef}
+                >
+                    {Object.entries(priorityLabels).map(([key, label]) => (
+                        <ListItem key={key} value={key}>
+                            {renderPriorityIcon(key as Priority)}
+                            <Text inline size="small">
+                                {label}
+                            </Text>
+                        </ListItem>
+                    ))}
+                </Dropdown>
+            )}
+
+            <Dropdown<Option>
+                button={
+                    <button
+                        {...elem('optionButton', {
+                            isSelected: dropdownStates.option,
+                        })}
+                        aria-label={`${option?.label} option button`}
+                        disabled={isOptionButtonDisabled}
+                    >
+                        <Text
+                            inline
+                            size="small"
+                            {...elem('optionButton--valueText')}
+                            {...(typeof children === 'string' && { title: children })}
+                        >
+                            {children}
+                        </Text>
+                        {option && (
+                            <Text
+                                {...elem('optionButton--optionText')}
+                                inline
+                                title={option?.label}
+                                size="small"
+                            >
+                                {option.label}
+                            </Text>
+                        )}
+                    </button>
+                }
+                additionalSelectProps={{
+                    onStateChange: (state) => toggleDropdown('option', state.isOpen),
+                }}
+                onChange={(newOption) => onChange(newOption)}
+                placement="bottom"
+                refElement={badgeRef}
+            >
+                <div {...elem('listHeadline')}>
+                    <Text inline size="small" title={optionListHeader}>
+                        {optionListHeader?.toUpperCase()}
+                    </Text>
+                </div>
+                {optionList?.map((opt) => (
+                    <ListItem key={opt.value} value={opt}>
+                        <Text inline size="small">
+                            {opt.label}
+                        </Text>
+                    </ListItem>
+                ))}
+            </Dropdown>
+
+            {onClose && (
+                <button
+                    {...elem('closeButton')}
+                    aria-label="closeButton"
```

Comment:
The label must be human-readable text`close button`

Reply:
missed it, will correct